Adds another input function that must return wether an element is sho…

gabrielci opened the pull request:
…wn or not, removing this type of logic from the renderBranch function

gabrielci committed:
Adds another input function that must return wether an element is shown or not, removing this type of logic from the renderBranch function

pauljonescodes commented on src/tree-root.tsx:
```diff
@@ -14,17 +14,15 @@ export type TreeRootProps<BranchType, LeafType> =
-      onToggle: () => void,
-    ) => ReactNode;
+    renderBranch: (item: BranchType) => ReactNode;
+    getIsToggled: (data: TreeData<BranchType, LeafType>) => boolean;
```
I would prefer this become a fully controlled element with this being passed in as data

gabrielci replied:
Added a `isToggled` property to `TreeData` and removed the whole `getIsToggled(...)` function, additionally, had to change the parameter of `renderBranch` to be the `tree` instead of just the `tree.data` so `isToggled` can be properly accessed on `renderBranch`

gabrielci committed:
Adds toggle as a prop reverting some of the changes of the last commit

- A new prop isToggled was added to the TreeData type, this is now used to show/hide the content of a branch
- Adjusted parameter for renderBranch to be the actual tree instead of just the tree.data so it can reference the new isToggled prop properly
